codex/node.nim

```diff
@@ -284,14 +286,25 @@ proc streamEntireDataset(
       except CatchableError as exc:
         trace "Error erasure decoding manifest", manifestCid, exc = exc.msg
 
-    self.trackedFutures.track(erasureJob())
+    jobs.add(erasureJob())
 
-  self.trackedFutures.track(self.fetchDatasetAsync(manifest, fetchLocal = false))
-    # prefetch task should not fetch from local store
+  jobs.add(self.fetchDatasetAsync(manifest))
+
+  # Monitor stream completion and cancel background jobs when done
+  proc monitorStream() {.async: (raises: []).} =
+    try:
+      await stream.join()
+    except CatchableError as exc:
+      warn "Stream failed", exc = exc.msg
```

Comment:
`stream.join` has the following signature:

```nim
proc join*(
    s: LPStream
): Future[void] {.async: (raises: [CancelledError], raw: true), public.} =
```

Thus, it can only raise `CancelledError`. Thus, wouldn't it be better to be more specific catching only the `CancelledError` and having a more precise message (like `Stream cancelled`)?

Reply:
Thats fine, this workers shouldn't allow anything to leak, so having a catch all exception is fine IMO.